java_gen/templates/custom/interface/OFMessage.java

```diff
@@ -0,0 +1,15 @@
+    
+    /**
+     * Compares the two messages for equality, ignoring the XID field.
+     *
+     * @param obj the other message to compare
+     * @return true if the messages are equal, ignoring the XID; false otherwise
+     */
+    boolean equalsIgnoreXid(Object obj);
+
+    /**
+     * Computes the hashcode of the message, ignoring the XID field.
+     *
+     * @return the hashcode of the message, ignoring the XID
+     */
+    int hashCodeIgnoreXid();
```

Comment:
I understand the rationale behind `equalsIgnoreXid`, but what is the usage model for `hashCodeIgnoreXid`? I'm asking because `hashCode` is rarely called by the user; it's usually implicitly called when you place the object in question in a Hash-based collection.

Since any existing collection will still always call `.hashCode` and not `.hashCodeIgnoreXid`, what's the value of exposing this method?